src/Parse/ParseQuery.php

```diff
@@ -623,6 +623,26 @@ public function withinGeoBox($key, $southwest, $northeast)
     }
 
     /**
+     * Add a constraint to the query that requires a particular key's
+     * coordinates be contained within a given polygon
+     *
+     * @param string        $key       The key of the ParseGeoPoint
+     * @param array         $points    Array of ParseGeoPoint
+     *
+     * @return ParseQuery Returns this query, so you can chain this call.
+     */
+    public function withinPolygon($key, $points)
+    {
+        $this->addCondition(
+            $key,
+            '$geoWithin',
+            ['$polygon' => $points]
+        );
+
+        return $this;
+    }
+
```

Comment:
@dplewis If you can make sure you have the following:
- Test with less than 3 points (should fail)
- Test with no points/empty array (should fail i think...)
- Test with at least 3 points (just to verify the min works)
- Test for returning an object that lies on the bounds, can have the same coords as one of the points for example
- Test for returning an object that lies beyond the bounds (fail, you may already have this though, if you do ignore this)
- Test for returning an object within bounds (pretty sure you have this already so you should be good)
- Test passing something other than a GeoPoint to withinPolygon (should fail)

That should tightly define what we will be expecting in terms of behavior from this additional query parameter in this sdk. There might be one or two more but that sums it up for now. Some of the above tests you already have included, and that's fine, it's just a summary of all the cases I can think of to account for.